sync/src/synchronizer/mod.rs

```diff
@@ -150,8 +150,8 @@ impl Synchronizer {
         }
     }
 
-    pub fn get_blocks_to_fetch(&self, peer: PeerIndex) -> Option<Vec<packed::Byte32>> {
-        BlockFetcher::new(self.clone(), peer).fetch()
+    pub fn get_blocks_to_fetch(&self, peer: PeerIndex, ibd: bool) -> Option<Vec<packed::Byte32>> {
```

Comment:
I really hate boolean parameter in function, it makes the place invoking the function hard to read. An enum or a new type is much better than a boolean.

Ex.

```
enum IBDState {
  IBD,
  NotIBD,
}

struct IBDState(bool);
impl IBDState {
  bool is_ibd(&self) -> { self.0 }
}
```